web-console/knikubevirt/expose-vm-as-a-service/expose-vm-as-a-service.md

```diff
@@ -0,0 +1,53 @@
+# Expose VM as a Network Service
+
+The user should be able to expose a VM as a network service. This means that they would be exposing some of the ports on its default interface to the cluster internally or to the outside world. In CNV 1.4, if we create a VM and do not identify an L2 network, our VM cannot be reached from the outside world nor it can be reached by other pods. In order for it to be actually useful, we need to bound it to a service and today we have to do it manually.
+
+## Expose From VM List
+
+![Expose From VM List](img/Expose_01.png)
+
+We added a new action called “Expose VM as Network service”. Clicking on that will open the expose as network popup. 
+
+![Pop up](img/Expose_02.png)
+
+![Pop up open types](img/Expose_03.png)
+
+The user can select between 3 network types:
+-  Cluster IP (Default) - Exposing the VM to inside the cluster 
+-  Load Balancer - Exposing the VM to the Internet 
+-  Node Port - not sure yet what is the purpose for that  
+
+![3 network types](img/Expose_04a.png)
+
+The user can choose between TCP or UDP protocol. 
+
+![3 network types](img/Expose_04_b.png)
+
+The name is the only must field in the menu. All other parts can be predefined, meaning they will have “default offer” using kubernetes capabilities. 
+
+
+![3 network types](img/Expose_05a.png)
+
+The Service is ready for use. The user can see it in the services list
+
+![3 network types](img/Expose_05b.png)
+
+## Expose From VM Page 
+
+![3 network types](img/Expose_06.png)
+
+We added the same action “Expose as Network service” to the actions menu.
+
+![3 network types](img/Expose_07.png)
```

Comment:
I would say we can get rid of this along with the next two images in the interest of trying to keep this doc on the shorter side. I think it's fine to show the action in the dropdown and then just mention that the user would go through the same flow. 

I do like the addition of the new design around showing Services on the Overview tab. @ranelim put up a PR [here](https://github.com/openshift/openshift-origin-design/pull/171) that could probably just be covered here rather than a separate link in the docs...what do you guys think?

Reply:
hey @lizsurette, do you mean merge the services PR that https://github.com/Ranelim created into this one?

Reply:
@yfrimanm Yes exactly...seems like a natural place for it to fit rather than be it's own separate section. What do you think @ranelim ?

Reply:
Hi, sorry for some reason I'm not getting email notifications for Github (no idea why) so Yifat just told me about it.

It does make sense since the services section in the overview is where the user lands after  "exposing VM as a service". But, during the call with the Openshift team, we kinda agreed (unofficially) that this services section to be a generic section for any resource that can be exposed as a service, like pods. 

I think that if that decision is still relevant, then we either need to make 'Expose [resource] as a service' a generic PR (on the Openshift design 4.0), or the Services section as a generic one.

If we intend to implement the services section only on VM, for some reason, then yeah, let's merge them